### ISSUE TITLE
`prefer-tacit`: regression on method calls

### ISSUE THREAD

**Issue by abrahamguo** [eslint-functional/eslint-plugin-functional#805]
## Bug Report
Regression introduced in #801:
`prefer-tacit` reports on `[''].filter(str => /a/.test(str))`.

### Expected behavior

No errors reported, since changing to `[''].filter(/a/.test)` causes a JavaScript error.

### Actual behavior

No reports by `prefer-tacit`.

**Comment by RebeccaStevens** [comment]
Looks like regex methods require a `this` reference to the regex itself, and thus without it things go wrong. I imagine this affects more built-in objects than just regex so let me know if you encounter other ones.

**Comment by RebeccaStevens** [comment]
Note: if you do want to use the tacit version, you can with help from `bind`.
e.g.

```ts
const regex = /a/;
[''].filter(regex.test.bind(regex));
```

**Comment by RebeccaStevens** [comment]
I added a new option `checkMemberExpressions` which is `false` by default. Setting it to `true` will enable the previous behavior; except now `.bind(...)`s are included in the suggestion.

**Comment by abrahamguo** [comment]
Perfect, thanks!
